bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/JsonParser.java

```diff
@@ -1249,7 +1252,7 @@ public State transition(StateMachine sm, char[] buff, int i, int count) throws J
                             }
                             break;
                         default:
-                            StateMachine.throwExpected("escaped characters");
+                            throw new JsonParserException("expected escaped characters");
```

Comment:
Shall we add a small test to cover this as well?

Reply:
Added with https://github.com/ballerina-platform/ballerina-lang/pull/40396/commits/909ceb0ff7ff7aca1c4f263117bb358f1d667498